iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -68,7 +70,15 @@ public async Task RunSampleAsync(TimeSpan sampleRunningTime)
             try
             {
                 await InitializeAndOpenClientAsync();
-                await Task.WhenAll(SendMessagesAsync(cts.Token), ReceiveMessagesAsync(cts.Token));
+
+                if (_transportType.Equals(TransportType.Http1))
+                {
+                    await TaskHelper.WhenAllFailFast(SendMessagesAsync(cts.Token), ReceiveMessagesAsync(cts.Token), CompleteReceivedMessagesAsync(cts.Token));

```

Comment:
Some amount of DRY violation here. You could create a list and initialize it with the 2 common tasks, then optionally add the `ReceiveMessagesAsync` task inside an if statement. Then after, use TaskHelper.WhenAllFailFast on the list.